File: Tests/Reqnroll.Formatters.Tests/Samples/Resources/pending/pending.cs
using Reqnroll;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CucumberMessages.CompatibilityTests.CCK.pending
{
    [Binding]
    internal class Pending
    {
        [Given("an implemented non-pending step")]
        public void GivenAnImplementedNonPendingStep()
        {
            //nop
        }

        [Given("an implemented step that is skipped")]
        public void GivenAnImplementedStepThatIsSkipped()
        {
            throw new ApplicationException("This step should not have been executed");
        }

        [Given("an unimplemented pending step")]
        public void GivenAnUnimplementedPendingStep()
        {
            throw new PendingStepException();
        }
    }
}

File: Tests/Reqnroll.Formatters.Tests/Samples/Resources/pending/pending.ts
import { Given } from '@cucumber/fake-cucumber'

Given('an implemented non-pending step', function () {
  // no-op
})

Given('an implemented step that is skipped', function () {
  // no-op
})

Given('an unimplemented pending step', function () {
  return 'pending'
})
